integ/emf_logger/logscript.py

```diff
@@ -0,0 +1,44 @@
+import time
+import sys
+import os
+import random
+import socket
+
+"""
+This script writes a single EMF payload to the TCP socket
+https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html
+"""
+
+# Define constants
+metricNamespace = "fluent-metrics"
+metricName = 'fluent-bit-integ-test-{:d}'.format(random.randint(0, 32767))
+metricValue = 100
+
+# pass the metric-name to a file in the shared volume for validation
+metricPath = os.getenv('EMF_METRIC_NAME_PATH')
+file = open(metricPath, "w")
+file.write(metricName)
+file.close()
+
+# EMF payload.
+emfPayload = ('{{"_aws":{{"Timestamp":{:d}000,"CloudWatchMetrics":[{{"Namespace":"{:s}","Dimensions":[["dimensionKey"]]'
+              ',"Metrics":[{{"Name":"{:s}"}}]}}]}},"dimensionKey":"dimensionValue","{:s}":{:d}}}\n').format(
+    int(time.time()), metricNamespace, metricName, metricName, metricValue)
+
+dataToSend = str.encode(emfPayload)
+
+# On Linux, connect to the Unix socket.
+# For Windows, we need to connect to a TCP socket running in Fluent-bit container.
+host = os.getenv('FLUENT_CONTAINER_IP')
```

Comment:
who/where sets this env var?

Reply:
This is set from the `run-integ.ps1` script which sets up the integration tests on Windows.